lib/vanagon/engine/base.rb

```diff
@@ -73,9 +73,15 @@ def ship_workdir(workdir)
         Vanagon::Utilities.rsync_to("#{workdir}/*", "#{@target_user}@#{@target}", @remote_workdir, @platform.ssh_port)
       end
 
-      def retrieve_built_artifact
-        FileUtils.mkdir_p("output")
-        Vanagon::Utilities.rsync_from("#{@remote_workdir}/output/*", "#{@target_user}@#{@target}", "output/", @platform.ssh_port)
+      def retrieve_built_artifact(artifacts_to_fetch, no_packaging)
+        output_path = 'output/'
+        FileUtils.mkdir_p(output_path)
+        unless no_packaging
+          artifacts_to_fetch << "#{@remote_workdir}/output/*"
+        end
+        artifacts_to_fetch.each do |path|
+          Vanagon::Utilities.rsync_from(path, "#{@target_user}@#{@target}", output_path, @platform.ssh_port)
```

Comment:
Are you expecting the `artifacts_to_fetch` list to include build metadata/etc files or should we automagically be pulling these in?

Reply:
I didn't modify what happens by default in this function, so I _think_ the build metadata/etc stuff was just included in the output/ dir on the builder right?

Reply:
I think those only get fetched due to the `output/*` in L80 though, so I think if `no_packaging` is true you won't get them

Reply:
Ah, I see. I'll have this function produce metadata files if no_packaging is true

Reply:
@underscorgan so, I took a look at what we do with the metadata file. And it turns out we only fetch a build_metadata file from output/ when we are generating an archive as the output: https://github.com/puppetlabs/vanagon/blob/0.15.1/lib/vanagon/platform.rb#L459L485

And even that is a duplicate. The save_manifest_json function here: https://github.com/puppetlabs/vanagon/blob/0.15.1/lib/vanagon/driver.rb#L136 saves the same information in ext (which it will still do during a no_packaging build)

Since no_packaging should identify that no packaging steps are going to be executed, it wouldn't make sense have generate_compiled_archive and no_packaging set to true.

With that information, should I still figure out making and grabbing the metadata file from output/ ?

Reply:
I was worried we wouldn't end up with those files, but since we do, I'm fine with this.